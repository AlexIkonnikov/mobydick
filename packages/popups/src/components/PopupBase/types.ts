import {TextStyle} from 'react-native';
import {TypographyProp} from '@npm/mobydick-typography';

import {IPopup} from '../../types';

export interface ITitlePopup {
  title: string;
  titleStyles?: TextStyle | TextStyle[];
  titleFont?: TypographyProp;
}

export interface IDescriptionTextPopup {
  descriptionText: string;
  descriptionStyles?: TextStyle | TextStyle[];
  descriptionFont?: TypographyProp;
}

export interface IContentTextProps {
  title?: string;
  titleStyles?: TextStyle | TextStyle[];
  titleFont?: TypographyProp;

  descriptionText?: string;
  descriptionStyles?: TextStyle | TextStyle[];
  descriptionFont?: TypographyProp;
}

export interface IPopupCloseIcon {
  onPress(): void;
}

export type IPopupProps = Pick<IPopup, 'overlayStyle'> & {
  onClose: () => void;
};
