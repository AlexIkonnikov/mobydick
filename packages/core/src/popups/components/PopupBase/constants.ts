enum Constants {
  testID = 'Pressable',
}

export default Constants;
