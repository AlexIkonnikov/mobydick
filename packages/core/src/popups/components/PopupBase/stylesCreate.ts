import {StyleSheet} from 'react-native';

import {IThemeContext} from '../../../styles/theme/types';

const stylesCreate = (theme: IThemeContext) => {
  const {colors} = theme;
  return StyleSheet.create({
    overlay: {
      position: 'absolute',
      height: '100%',
      width: '100%',
      backgroundColor: colors.BgOverlay,
      alignItems: 'center',
    },
  });
};

export default stylesCreate;
