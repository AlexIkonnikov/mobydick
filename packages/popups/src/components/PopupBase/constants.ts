enum Constants {
  testID = 'Pressable',
  accessibilityLabelActionSheetsItem = 'accessibilityLabelActionSheetsItem',
}

export default Constants;
