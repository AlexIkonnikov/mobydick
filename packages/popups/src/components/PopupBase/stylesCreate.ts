import {StyleSheet} from 'react-native';
import {IThemeContext} from '@npm/mobydick-styles';

const stylesCreate = (theme: IThemeContext) => {
  const {colors, spaces} = theme;
  return StyleSheet.create({
    overlay: {
      position: 'absolute',
      height: '100%',
      width: '100%',
      backgroundColor: colors.BgOverlay,
      alignItems: 'center',
      justifyContent: 'flex-end',
    },
    container: {
      backgroundColor: colors.BgSecondary,
      borderRadius: spaces.Space12,
      justifyContent: 'center',
      // alignItems: 'center',
      padding: spaces.Space20,
      marginVertical: spaces.Space20,
      marginHorizontal: spaces.Space6,
    },
  });
};

export default stylesCreate;
