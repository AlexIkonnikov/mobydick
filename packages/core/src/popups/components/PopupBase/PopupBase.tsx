import React, {FC, PropsWithChildren, useEffect} from 'react';
import {BackHandler} from 'react-native';

import useStyles from '../../../styles/theme/hooks/useStyles';
import Pressable from '../../../basic/components/Pressable/Pressable';

import Constants from './constants';
import {IPopupProps} from './types';
import stylesCreate from './stylesCreate';

const PopupBase: FC<PropsWithChildren<IPopupProps>> = ({
  onClose,
  children,
  overlayStyle,
}) => {
  const [styles] = useStyles(stylesCreate);

  useEffect(() => {
    const onBackPress = () => {
      onClose();
      return true;
    };

    BackHandler.addEventListener('hardwareBackPress', onBackPress);
    return () =>
      BackHandler.removeEventListener('hardwareBackPress', onBackPress);
  }, []);

  return (
    <Pressable
      style={[styles.overlay, overlayStyle]}
      testID={Constants.testID}
      onPress={event => {
        if (event.target === event.currentTarget) {
          onClose();
        }
      }}>
      {children}
    </Pressable>
  );
};

export default PopupBase;
