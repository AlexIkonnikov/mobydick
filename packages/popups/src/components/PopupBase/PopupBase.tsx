import React, {FC} from 'react';
import {Pressable} from '@npm/mobydick-core';
import {useStyles} from '@npm/mobydick-styles';

import Constants from './constants';
import {IPopupProps} from './types';
import stylesCreate from './stylesCreate';

const PopupBase: FC<IPopupProps> = ({onClose, children, overlayStyle}) => {
  const [styles] = useStyles(stylesCreate);

  return (
    <Pressable
      style={[styles.overlay, overlayStyle]}
      testID={Constants.testID}
      onPress={event => {
        if (event.target === event.currentTarget) {
          onClose();
        }
      }}>
      {children}
    </Pressable>
  );
};

export default PopupBase;
