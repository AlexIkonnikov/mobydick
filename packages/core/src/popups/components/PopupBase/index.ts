import PopupBase from './PopupBase';

export {PopupBase};

export * from './types';
