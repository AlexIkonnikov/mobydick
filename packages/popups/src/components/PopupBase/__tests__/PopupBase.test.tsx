import React, {useContext} from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import Constants from '@npm/mobydick-popups/src/components/PopupBase/constants';
import {renderHook} from '@testing-library/react-hooks';
import {PopupsContext, usePopups} from '@npm/mobydick-popups';

import PopupBase from '../PopupBase';

jest.mock('../../../hooks/usePopups');

describe('@npm/mobydick-popups/PopupBase', () => {
  afterEach(() => {
    jest.resetAllMocks();
    jest.clearAllMocks();
  });
  it('should renders correctly', () => {
    const popupContextMock = {
      closePopup: jest.fn(),
    };
    (usePopups as jest.Mock).mockReturnValue(popupContextMock);
    const id = '41710252-4116-46ba-b2e9-9ec0d34f9db8';
    const {toJSON, getByTestId} = render(
      <PopupBase id={id} Content={() => null} onClose={() => null} />,
    );
    const pressable = getByTestId(Constants.testID);

    fireEvent.press(pressable, {target: null, currentTarget: null});

    expect(toJSON()).toMatchSnapshot();
    expect(popupContextMock.closePopup).toHaveBeenCalledTimes(1);
    expect(popupContextMock.closePopup).toHaveBeenCalledWith(id);
  });
  it('should not fire onClose event', () => {
    const {result} = renderHook(() => useContext(PopupsContext));

    expect(result.current).toStrictEqual(
      expect.objectContaining({
        popups: [],
      }),
    );

    const closePopup = jest.spyOn(result.current, 'closePopup');

    const id = '41710252-4116-46ba-b2e9-9ec0d34f9db8';
    const {toJSON, getByTestId} = render(
      <PopupBase id={id} Content={() => null} onClose={() => null} />,
    );
    const pressable = getByTestId(Constants.testID);

    fireEvent.press(pressable, {target: {}, currentTarget: {}});

    expect(toJSON()).toMatchSnapshot();
    expect(closePopup).toHaveBeenCalledTimes(0);
  });
});
