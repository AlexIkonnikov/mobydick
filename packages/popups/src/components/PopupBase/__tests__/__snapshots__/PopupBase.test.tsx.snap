// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@npm/mobydick-popups/PopupBase should not fire onClose event 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(32, 36, 45, 0.45)",
        "height": "100%",
        "justifyContent": "flex-end",
        "position": "absolute",
        "width": "100%",
      },
      undefined,
    ]
  }
  testID="Pressable"
/>
`;

exports[`@npm/mobydick-popups/PopupBase should renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(32, 36, 45, 0.45)",
        "height": "100%",
        "justifyContent": "flex-end",
        "position": "absolute",
        "width": "100%",
      },
      undefined,
    ]
  }
  testID="Pressable"
/>
`;
